scrapi/harvesters/elife.py

```diff
@@ -0,0 +1,171 @@
+    jsonstr = resp.content.decode('utf-8')
+    jsonobj = json.loads(jsonstr)
+
+    files = [d['filename'] for d in jsonobj['files']]
+    return files
+
+
+def fetch_xml(xml_url, filename):
+    xml_text = requests.get(xml_url.format(filename)).content
+    xml = etree.fromstring(xml_text)
+    return xml
+
+
+class ELifeHarvester(XMLHarvester):
+    short_name = 'elife'
+    long_name = 'eLife Sciences'
+    url = 'http://elifesciences.org/'
+    DEFAULT_ENCODING = 'UTF-8'
+    record_encoding = None
+
+    namespaces = {}
+
+    MAX_ROWS_PER_REQUEST = 999
+    BASE_URL = 'https://api.github.com/repos/elifesciences/elife-articles/commits?'
+    BASE_COMMIT_URL = 'https://api.github.com/repos/elifesciences/elife-articles/commits/{}'
+    BASE_DATA_URL = 'https://raw.githubusercontent.com/elifesciences/elife-articles/master/{}'
+
+    def harvest(self, start_date=None, end_date=None):
+        start_date = start_date or datetime.date.today() - datetime.timedelta(settings.DAYS_BACK)
+        end_date = end_date or datetime.date.today()
+
+        shas = fetch_commits(self.BASE_URL, start_date.isoformat(), end_date.isoformat())
+
+        files = []
+        for sha in shas:
+            files = files + fetch_file_names(self.BASE_COMMIT_URL, sha)
+            files = list(set(files))
+
+        xml_records = []
+        for file in files:
+            xml_records.append(fetch_xml(self.BASE_DATA_URL, file))
+
+        return [
+            RawDocument({
+                'filetype': 'xml',
+                'source': self.short_name,
+                'doc': etree.tostring(record),
+                'docID': record.xpath('//article-id[@*]')[0].text,
+            }) for record in xml_records
+        ]
+
+    schema = {
+        'uris': {
+            'canonicalUri': ('//article-id/node()', compose('http://dx.doi.org/10.7554/eLife.{}'.format,
+                                                            single_result)),
+        },
+        'contributors': ('//article-meta/contrib-group/contrib/name/*[not(self::suffix)]/node()', elife_name_parser),
+        'providerUpdatedDateTime': ('//article-meta/pub-date[@publication-format="electronic"]/*/node()',
+                                    compose(datetime_formatter, elife_date_parser)),
+        'title': ('//article-meta/title-group/article-title//text()', title_parser),
+        'description': ('//abstract[not(@abstract-type="executive-summary")]/p//text()', description_parser),
+        'publisher': {
+            'name': ('//publisher-name/node()', single_result)
+        },
+        'subjects': '//article-meta/article-categories/descendant::text()',
+        'freeToRead': {
+            'startDate': ('//article-meta/pub-date[@publication-format="electronic"]/*/node()',
+                          elife_date_parser)
+        },
+        'tags': '//kwd/text()',
+        'otherProperties': build_properties(
+                ('license', ('//permissions/license/license-p/ext-link/text()', single_result))
```

Comment:
Still having trouble getting it to accept the license as part of the schema - I listed it as rights in other properties for now (which is the case for most of the harvesters currently)